src/hooks/transactions/useSignTransactions.tsx

```diff
@@ -5,185 +5,228 @@ import {
+    callbackRoute: string = ''
+  ) => {
+    const urlParams = { [walletSignSession]: sessionId };
+    const callbackUrl = `${window.location.origin}${callbackRoute}`;
+    const buildedCallbackUrl = buildReplyUrl({ callbackUrl, urlParams });
+
+    provider.signTransactions(transactions, {
+      callbackUrl: encodeURIComponent(buildedCallbackUrl)
+    });
   };
 
-  async function signTransactionsWithProvider() {
+  const signTransactionsWithProvider = async () => {
+    const {
+      sessionId,
+      transactions,
+      callbackRoute,
+      customTransactionInformation
+    } = transactionsToSign!;
+    const { redirectAfterSign } = customTransactionInformation;
+    const redirectRoute = callbackRoute || window.location.pathname;
+    const isCurrentRoute = window.location.pathname.includes(redirectRoute);
+    const shouldRedirectAfterSign = redirectAfterSign && !isCurrentRoute;
+
     try {
-      const {
-        sessionId,
-        transactions,
-        callbackRoute,
-        customTransactionInformation: { redirectAfterSign }
-      } = transactionsToSign!;
-      const redirectRoute = callbackRoute || window.location.pathname;
-      if (transactions?.length) {
-        const initialized = await provider.init();
-        if (!initialized) {
-          return;
-        }
-        try {
-          const signedTransactions: Transaction[] =
-            await provider.signTransactions(transactions);
-
-          const signingDisabled =
-            !signedTransactions ||
-            (signedTransactions &&
-              Object.keys(signedTransactions).length !== transactions?.length);
-          if (!signingDisabled && signedTransactions) {
-            dispatch(
-              moveTransactionsToSignedState({
-                sessionId,
-                status: TransactionBatchStatusesEnum.signed,
-                transactions: Object.values(signedTransactions).map((tx) =>
-                  parseTransactionAfterSigning(tx)
-                )
-              })
-            );
-
-            if (
-              redirectAfterSign &&
-              !window.location.pathname.includes(redirectRoute)
-            ) {
-              window.location.href = redirectRoute;
-            }
-          }
-        } catch (err) {
-          const errorMessage =
-            (error as unknown as Error)?.message ||
-            (error as string) ||
-            'error signing transaction';
-
-          console.error('error signing transaction', errorMessage);
-          onCancel(errorMessage, sessionId);
-        }
+      const isProviderInitialized = await provider.init();
+
+      if (!isProviderInitialized) {
+        return;
       }
-    } catch (err) {
+    } catch (error) {
       const errorMessage =
         (error as unknown as Error)?.message ||
         (error as string) ||
-        'error signing transaction';
-      console.error('error signing transaction', errorMessage);
+        errorsMessages.PROVIDER_NOT_INTIALIZED;
```

Comment:
there's a little thing that I've added on useSignWithLedger, but seems to be missing here. What do you say if we add this error to redux, so it can be read?

  dispatch(setSignTransactionsError((err as unknown as Error).message));